<launch>
  <!-- Arguments -->
  <arg name="dist_follow_wall" default="0.15" />
  <arg name="speed_max" default="50.0" />
  <arg name="rviz" default="false" />
  <!-- Parameters -->
  <param name="dist_follow_wall" value="$(arg dist_follow_wall)" />
  <param name="speed_max" value="$(arg speed_max)" />
  <param name="rviz" value="$(arg rviz)" />

  <!-- Start the nodes -->
  <node name="crash_node" pkg="auto_car_ctrl" type="crash_test" />
  <node name="lidar_process_node" pkg="auto_car_ctrl" type="lidar_process" />
  <node name="cmd_vel_node" pkg="auto_car_ctrl" type="cmd_vel_node" />
  
  <group if="$(eval rviz == true)">
    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find auto_car_ctrl)/rviz/auto_car_ctrl_rviz.rviz" />
  </group>
  
  <!-- Start the arduino communication -->
  <arg name="arduino_port" default="ttyACM0" />
  <node name="arduino_comm" pkg="rosserial_python" type="serial_node.py" args="/dev/$(arg arduino_port)" />
  
  <!-- Start the lidar -->
  <include file="$(find ydlidar_ros)/launch/X4.launch" />
</launch>
